src/post_processing.jl

```diff
@@ -446,24 +499,38 @@ function write_powerflow_solution!(
             bus_name = temp_bus_map[bus_number]
             bus = PSY.get_component(PSY.ACBus, sys, bus_name)
             ix = get_bus_lookup(data)[bus_number]
-            if bus.bustype == PSY.ACBusTypes.REF
+            bustype = data.bus_type[ix, time_step] # may not be the same as bus.bustype!
```

Comment:
This for-loop has some code duplication. Is it because the above for loop will be removed?